resources/osl_awstats_site.rb

```diff
@@ -22,16 +34,51 @@
   template "/etc/awstats/awstats.#{new_resource.name}.conf" do
     cookbook 'osl-resources'
     source 'awstats_site.conf.erb'
-    filename = new_resource.log_file
-    filename.concat(new_resource.date_format) if new_resource.append_date
-    filename.concat(new_resource.log_suffix) if new_resource.append_suffix
+
+    # Determine the log file parameter
+    log_file = ''
+    if new_resource.use_osl_mirror
+      log_file = '/usr/share/awstats/tools/logresolvemerge.pl '
+      dir_base = '/var/lib/awstats/logs/ftp-'
+
+      filename = ''
+      filename.concat('_ftp') if new_resource.vsftp_logs
+      filename.concat('/')
+      filename.concat(new_resource.log_file)
+      filename.concat(new_resource.date_format) if new_resource.append_date
+      filename.concat(new_resource.log_ext)
+
+      mirrors = %w(
+        osl
+        chi
+        nyc
+      )
+
+      log_file.concat(mirrors.map { |k| "#{dir_base}#{k}#{filename}" }.join(' '))
+      log_file.concat(' |')
+    else
+      log_file = new_resource.log_file
+      log_file.concat(new_resource.log_ext)
+    end
+
+    # Determine the site domain parameter
+    site_domain = new_resource.site_domain
+    site_domain = site_domain.join(' ') if site_domain.is_a?(Array)
+
+    # Determine the host aliases parameter
+    host_aliases = new_resource.host_aliases
+    host_aliases = host_aliases.join(' ') if host_aliases.is_a?(Array)
+
+    # Determine the only files parameter
+    only_files = new_resource.only_files
+    only_files = only_files.join(' ') if only_files.is_a?(Array)
```

Comment:
It looks like this might be better as a library helper method instead. I'd move the logic there and then just call it here.